pkg/gadgets/trace/network/tracer/tracer.go

```diff
@@ -333,3 +346,71 @@ func (t *Tracer) Close() {
 	}
 	t.networkGraphMapObjects.Close()
 }
+
+// --- Registry changes
+// TODO: This can be optimized a lot after using NewInstance() for everything
+
+func (g *GadgetDesc) NewInstance() (gadgets.Gadget, error) {
+	tracer := &Tracer{
+		attachments: make(map[uint64]*attachment),
+		cbmap:       make(map[uint64]func(ev *types.Event)),
+	}
+	return tracer, nil
+}
+
+func (t *Tracer) Init(gadgetCtx gadgets.GadgetContext) error {
+	// Load the eBPF map
+	t.gadgetCtx = gadgetCtx
+	specMap, err := loadGraphmap()
+	if err != nil {
+		return fmt.Errorf("failed to load asset: %w", err)
+	}
+	if err := specMap.LoadAndAssign(&t.networkGraphMapObjects, &ebpf.CollectionOptions{}); err != nil {
+		return fmt.Errorf("failed to load ebpf program: %w", err)
+	}
+	return nil
+}
+
+func (t *Tracer) Start() error {
+	ctx := t.gadgetCtx.Context()
+	for {
+		if ctx.Err() != nil {
+			return nil
+		}
+		// Pop, but we don't need the results as we're handing the events over to the callback in cbMap
+		_, err := t.Pop()
+		if err != nil {
+			t.gadgetCtx.Logger().Debugf("start() returned with: %v", err)
+			return nil
+		}
+	}
+}
+
+func (t *Tracer) Stop() {
+}
+
+func (t *Tracer) SetEventHandler(handler any) {
+	nh, ok := handler.(func(ev *types.Event))
+	if !ok {
+		panic("event handler invalid")
+	}
+	t.eventCallback = nh
+}
+
+func (t *Tracer) AttachContainer(container *containercollection.Container) error {
+	err := t.Attach(container.Pid)
+	if err != nil {
+		return err
+	}
+	t.cbmapMutex.Lock()
+	t.cbmap[container.Netns] = func(ev *types.Event) {
+		ev.SetContainerInfo(container.Podname, container.Namespace, container.Name)
+		t.eventCallback(ev)
+	}
```

Comment:
We don't need to replace the callback for a given `netns` every time we attach a new container that uses that `netns`. 

In addition:

1. It shouldn't enrich the event with the container name as we are unsure if more containers are using the same 
`netns`.
2. It shouldn't enrich the event if the container runs in the host network namespace.